### ISSUE TITLE
Google-free maps

### ISSUE THREAD

**Issue by Matt-Yorkley** [openfoodfoundation/openfoodnetwork#5542]
I'm adding this epic in the backlog to track any bits people do to move towards OFN being independent of Google Maps. Contributions welcome!

We're making a big leap forwards with this PR: https://github.com/openfoodfoundation/openfoodnetwork/pull/5398

There are still various bits that would need to be done before we can start migrating our current production servers to alternate providers.

- [ ] Clarify which other parts of the app depend on Google Maps JS (there are some other parts) for geocoding in javascript
- [ ] Make some alternative javascript services for geocoding, via some new API endpoints that use the Geocoder gem (which is already used elsewhere in the app, and can be easily configured with alternate providers)
- [ ] Improve map page searchbar functionality. The current OSM implementation only searches by names of OFN enterprises, not locations. This would need require the new non-google geocoding service, and some simple method calls to recenter the map on the found location.
- [ ] OSM map is totally blank (grey) when no enterprises are present. Maybe we could show the map and a message?

**Comment by sigmundpetersen** [comment]
Could/should #3179 be attached to this work? (Upgrading the geocoder dependency)

**Comment by diegoamc** [comment]
Hi!

What is the status of this issue? I would like to help if possible.

Is there a task list on what's left to be done?

**Comment by RachL** [comment]
Hi @Zoyolin thanks for your comment 🤗 

Gmaps was used first as 11 years ago in Australia (when OFN was launched)  OSM was not accurate / used.

Our community would love to have OSM working now but as you can see there is still a lot to do, so unless a volunteer picks this works up, we need to find funding, which is a long process.

**Comment by Zoyolin** [comment]
Thanks @RachL. Sorry I didn't realise how global OFN is and how much effort has already been put together to support OSM. I realised Ireland and Belgium's default is already OSM from leaflet. Indeed Australia, UK, France and USA are on ABC's map from Google.

**Comment by RachL** [comment]
@Zoyolin yes it's a setting you can access as super admin in `/admin/contents/edit`

But main issue with open street map currently is that the implementation let you search for enterprises, not location (unless you use the scroll). So if you don't know the enterprise name, the search is a bit difficult (although the address of the enterprise shows up in results as well if you type a city for example).

**Comment by filipefurtad0** [comment]
> yes it's a setting you can access as super admin at the bottom of the page /admin/contents/edit

An additional hint on this: it seems that enabling this option breaks the map around the first step of the enterprise registration process, at `/register`:

![image](https://github.com/openfoodfoundation/openfoodnetwork/assets/49817236/d1eb9d44-d799-47a6-a1f3-8b65d38b95dc)

**Comment by dacook** [comment]
I’m not sure of the exact status of Open Street Maps, but it appears at least mostly implemented, so we should probably document it for instance setup. I added a clue in the ofn-install wiki here, but more detail is needed: https://github.com/openfoodfoundation/ofn-install/wiki/Maps
But also there should be something in the super-admin guide too: https://ofn-user-guide.gitbook.io/ofn-super-admin-guide/ofn-platform-configuration/content

**Comment by RachL** [comment]
some info in this thread: https://openfoodnetwork.slack.com/archives/CG7NJ966B/p1701783880370229
